pkg/sql/sqlbase/structured.go

```diff
@@ -1868,6 +1868,30 @@ func (desc *TableDescriptor) validateCrossReferences(
 		}
 	}
 	// TODO(dan): Also validate SharedPrefixLen in the interleaves.
+
+	// Validate the all types present in the descriptor exist.
+	typeMap := make(map[ID]*TypeDescriptor)
+	getType := func(id ID) (*TypeDescriptor, error) {
+		if typeDesc, ok := typeMap[id]; ok {
+			return typeDesc, nil
+		}
+		typeDesc, err := GetTypeDescFromID(ctx, txn, codec, id)
```

Comment:
wonder if we need to batch fetch these at one point.

Reply:
Yeah jordan brought up on a different PR to pass in an interface here rather than a txn to be smarter about the lookups done.